rails/app/controllers/portal/clazzes_controller.rb

```diff
@@ -365,8 +365,11 @@ def info
 
   def external_report
     portal_clazz = Portal::Clazz.find(params[:id])
+    authorize portal_clazz
     report = ExternalReport.find(params[:report_id])
-    next_url = report.url_for_class(portal_clazz, current_visitor, request.protocol, request.host_with_port)
+    additional_params = {}
+    additional_params[:researcher] = 'true' unless user.has_full_access_to_student_data?(offering.clazz)
```

Comment:
I still think it would be cleaner to add the `researcher=true` to the url that calls this method via the researcher class UI than add it this way.  The researcher flag does not give them access but just tells the report to ask for researcher access - this method seems like it would be brittle to future changes in `has_full_access_to_student_data?`.